### PR TITLE
problem: czmq doesn't support new scatter/gather socket types

### PR DESCRIPTION
**Pull request by somdoron**

### CODE REVIEW

**Comment by sappo** [comment]
interesting, is there an documentation regarding that sockets?

**Comment by somdoron** [comment]
not yet, just created it today on libzmq. It works the same as push/pull but thread safe and single frame.

File: api/zsock.api [hunk 1]
```diff
@@ -126,6 +126,16 @@
         <argument name = "endpoint" type = "string" />
     </constructor>
 
+    <constructor name = "new gather" state = "draft">
+        Create a GATHER socket. Default action is bind.
+        <argument name = "endpoint" type = "string" />
+    </constructor>
+
+    <constructor name = "new scatter" state = "draft">
+        Create a GATHER socket. Default action is connect.
```
Comment (sappo) [inline api/zsock.api]: This should be SCATTER instead of GATHER, right?

**Comment by taotetek** [comment]
@somdoron This is wonderful. I'm going to add support for these to my rsyslog plugins when building against a czmq version that has them.

**Comment by sappo** [comment]
That's neat :+1:

File: api/zsock.api (hunk 1 above)
Comment (somdoron) [inline api/zsock.api]: yes, fixing that.

Comment (hitstergtd) [inline api/zsock.api]: @sappo Oops I just saw this! I went and pulled @somdoron's latest, should have let you do it. :(

Comment (sappo) [inline api/zsock.api]: @hitstergtd not at all, the faster we merge correct PRs the better. Even though I love pushing that green button :P

Comment (hitstergtd) [inline api/zsock.api]: @sappo - hah, thanks! I don't get to push that green button often enough, so have itchy trigger finger when it comes to Pulls. :)

**Comment by bluca** [comment]
This is really awesome stuff!

File: api/zsock.api (hunk 1 above)
Comment (sappo) [inline api/zsock.api]: @hitstergtd I can relate ;)

**Pull request merged**